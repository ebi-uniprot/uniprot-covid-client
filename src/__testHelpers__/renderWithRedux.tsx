/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
