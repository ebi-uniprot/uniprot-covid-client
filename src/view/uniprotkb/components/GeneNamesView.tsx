import React, { Fragment } from 'react';
import { v1 } from 'uuid';
import { InfoList } from 'franklin-sites';
import { GeneNamesData } from '../../../model/uniprotkb/sections/NamesAndTaxonomyConverter';
import { NameWithEvidence } from './ProteinNamesView';
import { ValueWithEvidence } from '../../../model/types/modelTypes';

export const geneAlternativeNamesView = (
  alternativeNames: ValueWithEvidence[],
  firstComma = true
) => {
  return (
    <Fragment>
      {firstComma && ', '}
      {alternativeNames
        .map<React.ReactNode>(altName => (
          <NameWithEvidence data={altName} key={altName.value} />
        ))
        .reduce((prev, curr) => [prev, ', ', curr])}
    </Fragment>
  );
};

const GeneNamesView: React.FC<{
  geneNamesData: GeneNamesData;
  isCompact?: boolean;
}> = ({ geneNamesData, isCompact = false }) => (
  <Fragment>
    {geneNamesData.map(geneNames => {
      const infoData = [
        {
          title: 'Name',
          content: geneNames.geneName && (
            <Fragment>
              <NameWithEvidence data={geneNames.geneName} />
            </Fragment>
          ),
        },
      ];
      if (geneNames.synonyms) {
        infoData.push({
          title: 'Synonyms',
          content: (
            <Fragment>
              {geneAlternativeNamesView(geneNames.synonyms, false)}
            </Fragment>
          ),
        });
      }
      if (geneNames.orfNames) {
        infoData.push({
          title: 'ORF names',
          content: (
            <Fragment>
              {geneAlternativeNamesView(geneNames.orfNames, false)}
            </Fragment>
          ),
        });
      }
      if (geneNames.orderedLocusNames) {
        infoData.push({
          title: 'Ordered locus names',
          content: (
            <Fragment>
              {geneAlternativeNamesView(geneNames.orderedLocusNames, false)}
            </Fragment>
          ),
        });
      }
      return (
        <InfoList
          infoData={infoData}
          key={geneNames.geneName ? geneNames.geneName.value : v1()}
          isCompact={isCompact}
          highlightFirstItem={isCompact}
        />
      );
    })}
  </Fragment>
);

export default GeneNamesView;
