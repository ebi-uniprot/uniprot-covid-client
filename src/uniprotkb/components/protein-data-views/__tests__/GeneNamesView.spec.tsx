import React from 'react';
import GeneNamesView from '../GeneNamesView';
import renderWithRedux from '../../../../shared/__test-helpers__/RenderWithRedux';
import GeneNamesUIDataJson from './__mocks__/geneNamesUIData.json';

describe('GeneNames', () => {
  test('should render gene_names', () => {
    const { asFragment } = renderWithRedux(
      <GeneNamesView geneNamesData={GeneNamesUIDataJson} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
