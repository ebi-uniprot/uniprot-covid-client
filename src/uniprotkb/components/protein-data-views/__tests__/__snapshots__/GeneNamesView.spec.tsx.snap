// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneNames should render gene_names 1`] = `
<DocumentFragment>
  <div
    class="info-list"
  >
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          Name
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        some Gene 
        <button
          aria-controls="<UUID>"
          aria-expanded="false"
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          type="button"
        >
          <svg
            fill="currentColor"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              d="M10 0H0v15l5-5 5 5V0z"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </button>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
          id="<UUID>"
        />
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          Synonyms
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        some Syn 
        <button
          aria-controls="<UUID>"
          aria-expanded="false"
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          type="button"
        >
          <svg
            fill="currentColor"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              d="M10 0H0v15l5-5 5 5V0z"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </button>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
          id="<UUID>"
        />
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          ORF names
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        some orf 
        <button
          aria-controls="<UUID>"
          aria-expanded="false"
          class="evidence-tag svg-colour-reviewed"
          data-testid="evidence-tag-trigger"
          type="button"
        >
          <svg
            fill="currentColor"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              d="M10 0H0v15l5-5 5 5V0z"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 publication
          </span>
        </button>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
          id="<UUID>"
        />
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          Ordered locus names
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        some locus 
        <button
          aria-controls="<UUID>"
          aria-expanded="false"
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          type="button"
        >
          <svg
            fill="currentColor"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              d="M10 0H0v15l5-5 5 5V0z"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </button>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
          id="<UUID>"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
